fs/remote/blob.go

```diff
@@ -281,89 +310,99 @@ func (b *blob) fetchRange(allData map[region]io.Writer, opts *options) error {
 		return nil
 	}
 
-	// Fetcher can be suddenly updated so we take and use the snapshot of it for
-	// consistency.
-	b.fetcherMu.Lock()
-	fr := b.fetcher
-	b.fetcherMu.Unlock()
-
-	// request missed regions
-	var req []region
-	fetched := make(map[region]bool)
-	for reg := range allData {
-		req = append(req, reg)
-		fetched[reg] = false
-	}
-	ctx, cancel := context.WithTimeout(context.Background(), b.fetchTimeout)
-	defer cancel()
-	mr, err := fr.fetch(ctx, req, true, opts)
-	if err != nil {
-		return err
-	}
-	defer mr.Close()
+	// We build a key based on regions we need to fetch and pass it to singleflightGroup.Do(...)
+	// to block simultaneous same requests. once the request is finished and the data is ready,
+	// all blocked callers will be unblocked and that same data will be returned by all blocked callers.
+	key := makeSyncKey(allData)
+	_, err, _ := b.fetchedRegionGroup.Do(key, func() (interface{}, error) {
```

Comment:
Writes to `allData map[region]io.Writer` need to occur even for blocked callers otherwise `ReadAt` will return invalid contents. 
Blocked callers should try to acquire contents from the cache after resuming. If they can't get the contents from the cache, they make requests to the registry.

Reply:
Done.